gravitee-apim-rest-api/gravitee-apim-rest-api-model/src/main/java/io/gravitee/rest/api/model/settings/PortalNext.java

```diff
@@ -57,5 +55,59 @@ public static class Banner {
 
         @ParameterKey(Key.PORTAL_NEXT_BANNER_CONFIG_ENABLED)
         private Boolean enabled;
+
+        private PrimaryButton primaryButton;
+
+        private SecondaryButton secondaryButton;
+
+        public Banner() {
+            super();
+            primaryButton = new PrimaryButton();
+            secondaryButton = new SecondaryButton();
+        }
+
+        @Data
+        @AllArgsConstructor
+        @NoArgsConstructor
+        @JsonIgnoreProperties(ignoreUnknown = true)
+        public static class PrimaryButton {
+
+            @ParameterKey(Key.PORTAL_NEXT_BANNER_PRIMARY_BUTTON_LABEL)
+            private String label;
+
+            @ParameterKey(Key.PORTAL_NEXT_BANNER_PRIMARY_BUTTON_TARGET)
+            private String target;
+
+            @ParameterKey(Key.PORTAL_NEXT_BANNER_PRIMARY_BUTTON_ENABLED)
+            private Boolean enabled;
+
+            @ParameterKey(Key.PORTAL_NEXT_BANNER_PRIMARY_BUTTON_TYPE)
+            private String type;
+
+            @ParameterKey(Key.PORTAL_NEXT_BANNER_PRIMARY_BUTTON_VISIBILITY)
+            private String visibility;
+        }
+
+        @Data
+        @AllArgsConstructor
+        @NoArgsConstructor
+        @JsonIgnoreProperties(ignoreUnknown = true)
+        public static class SecondaryButton {
+
+            @ParameterKey(Key.PORTAL_NEXT_BANNER_SECONDARY_BUTTON_LABEL)
+            private String label;
+
+            @ParameterKey(Key.PORTAL_NEXT_BANNER_SECONDARY_BUTTON_TARGET)
+            private String target;
+
+            @ParameterKey(Key.PORTAL_NEXT_BANNER_SECONDARY_BUTTON_ENABLED)
+            private Boolean enabled;
+
+            @ParameterKey(Key.PORTAL_NEXT_BANNER_SECONDARY_BUTTON_TYPE)
+            private String type;
+
+            @ParameterKey(Key.PORTAL_NEXT_BANNER_SECONDARY_BUTTON_VISIBILITY)
+            private String visibility;
+        }
```

Comment:
Why did you decide to create 2 identical but different classes?
I know that they differ in `@ParameterKey`
But can we reuse them?

Instead of 
`PORTAL_NEXT_BANNER_SECONDARY_BUTTON_LABEL` => `PORTAL_NEXT_BANNER_BUTTON_LABEL` 
etc.

Also in yml they are same objects:
`                primaryButton:
                  $ref: "#/components/schemas/BannerButton"
                secondaryButton:
                  $ref: "#/components/schemas/BannerButton"
`

So it makes sense to make them same in Java as well.

Reply:
It's not possible when using parameters. We want a specific string or boolean value to be converted from what could be defined in the `gravitee.yml` to specific attributes within the Banner class. This can work for the REST response, but when loading from specific parameters, it is not possible.